benchmarks/datasets/wikitext2_data.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+
+import io
+import warnings
+
+import torch
+from torch.utils.data import Dataset
+import torchtext
+from torchtext.data.utils import get_tokenizer
+from torchtext.utils import download_from_url, extract_archive
+from torchtext.vocab import build_vocab_from_iterator
+
+
+class SyntheticLMDataset(Dataset):
+    """
+    Dataset to benchmark a translation like seq2seq task.
```

Comment:
This class should have been deleted (have the change in a followup PR) since we don't use this class anymore. Sorry for the confusion!
I have the file organization as follows:
- datasets/wikitext2_data
- datasets/multi30k_data
It's true that a dataset could be used for multiple models in which case we could still share the common download logic but have a different collate_fn as needed for the dataloader. I imagine we'd have functions `get_real_dataloaders_lm` and `get_real_dataloaders_nmt`. I think we can make this change once we have this use case.

Reply:
I just remembered why I started off with a class structure. Would it make more sense to specify the configs as class attributes instead of returning a dict structure? This would not only allow separate namespaces but also allow us to refer to configurations in a more object oriented fashion e.g BenchmarkConfig.batch_size as opposed to benchmark_config["batch_size"]. Also a smaller chance of typo errors. WDYT?

cc @msbaines @blefaudeux since we are deciding the structure for all benchmarks going forward.